units/spawnhost_modify.go

```diff
@@ -49,56 +90,32 @@ func makeSpawnhostModifyJob() *spawnhostModifyJob {
 func NewSpawnhostModifyJob(h *host.Host, changes host.HostModifyOptions, ts string) amboy.Job {
 	j := makeSpawnhostModifyJob()
 	j.SetID(fmt.Sprintf("%s.%s.%s", spawnhostModifyName, h.Id, ts))
-	j.HostID = h.Id
 	j.ModifyOptions = changes
+	j.CloudHostModification.HostID = h.Id
 	return j
 }
 
 func (j *spawnhostModifyJob) Run(ctx context.Context) {
 	defer j.MarkComplete()
 
-	if j.env == nil {
-		j.env = evergreen.GetEnvironment()
+	// Setting the base field is for temporary backward compatibility with
+	// pending jobs already stored in the DB.
+	// TODO (EVG-16066): remove this check once all old versions of the job are
+	// complete.
+	if j.HostID != "" {
+		j.CloudHostModification.HostID = j.HostID
 	}
 
-	var err error
-	if j.host == nil {
-		j.host, err = host.FindOneByIdOrTag(j.HostID)
-		if err != nil {
-			j.AddError(err)
-			return
+	if err := j.CloudHostModification.modifyHost(ctx, func(mgr cloud.Manager, h *host.Host, user string) error {
```

Comment:
this if statement is a bit unclear to read; can we set the function as a variable instead and then just call it here?

Reply:
Sure, done. I changed the other places that do this as well.